File: spikes/009-ai-ide-extensions/workspace/sample-code/test_completions.py
"""
Sample Python file for testing AI code completions.

Test scenarios:
1. Function completion - type 'def calculate_' and see suggestions
2. Docstring generation - add docstring to existing function
3. Error handling - request try/except blocks
4. Type hints - request type annotations
"""

from dataclasses import dataclass
from pathlib import Path
from typing import Optional


@dataclass
class User:
    """Represents a user in the system."""
    id: int
    name: str
    email: str
    active: bool = True


def get_user_by_id(user_id: int, users: list[User]) -> Optional[User]:
    """Find a user by their ID."""
    for user in users:
        if user.id == user_id:
            return user
    return None


def process_user_data(filepath: Path) -> list[User]:
    """
    Load and process user data from a file.

    Test: Ask AI to add error handling and validation.
    """
    # TODO: Implement file reading
    # TODO: Add validation
    # TODO: Handle missing fields
    pass


# Test: Ask AI to complete this function
def calculate_statistics(numbers: list[float]):
    """Calculate basic statistics for a list of numbers."""
    # Let AI complete: mean, median, std_dev, min, max
    pass


# Test: Ask AI to generate tests for the User class
# Test: Ask AI to add logging to process_user_data
# Test: Ask AI to refactor get_user_by_id to use dict lookup

File: spikes/009-ai-ide-extensions/workspace/sample-code/test_completions.ts
/**
 * Sample TypeScript file for testing AI code completions.
 *
 * Test scenarios:
 * 1. Interface completion - add properties to interfaces
 * 2. Generic type completion - implement generic functions
 * 3. Async/await patterns - complete async functions
 * 4. Error handling - add try/catch blocks
 */

interface User {
  id: number;
  name: string;
  email: string;
  createdAt: Date;
  // Test: Ask AI to add more fields
}

interface ApiResponse<T> {
  data: T;
  status: number;
  message: string;
}

// Test: Ask AI to complete this function with proper error handling
async function fetchUser(userId: number): Promise<ApiResponse<User>> {
  // TODO: Implement fetch with error handling
  // TODO: Add retry logic
  // TODO: Add timeout
  throw new Error("Not implemented");
}

// Test: Ask AI to complete this validation function
function validateEmail(email: string): boolean {
  // Let AI complete the regex and validation logic
  return false;
}

// Test: Ask AI to implement this caching decorator
function memoize<T extends (...args: unknown[]) => unknown>(fn: T): T {
  // Let AI implement memoization
  return fn;
}

// Test: Ask AI to generate a React component from this interface
interface ButtonProps {
  label: string;
  onClick: () => void;
  variant?: "primary" | "secondary" | "danger";
  disabled?: boolean;
}

// Test: Ask AI to complete this class
class UserService {
  private users: Map<number, User> = new Map();

  // Let AI implement CRUD operations
}

// Test: Ask AI to add JSDoc comments to all functions above
// Test: Ask AI to convert UserService to functional approach
// Test: Ask AI to add zod schema validation for User interface
